client/src/controllers/TeleportController.ts

```diff
@@ -0,0 +1,40 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class TeleportController extends Controller<HTMLTemplateElement> {
+  static values = {
+    target: { default: 'body', type: String },
+  };
+
+  declare readonly targetValue: string;
+  declare targetContainer: HTMLElement;
+
+  /**
+   * It copies and appends the template content inside the body target
+   *
+   * @example
+   * <template
+   *    data-controller="w-teleport"
+   *    data-w-teleport-target-value="dialog_root_selector"
+   * >
+   *  <div>Some content</div>
+   * </template>
+   */
+  connect() {
+    this.targetContainer = document.querySelector(
+      this.targetValue,
+    ) as HTMLElement;
+
+    if (!this.targetContainer) {
+      throw new Error('Target container not found');
+    }
+
+    this.append();
+  }
+
+  append() {
```

Comment:
Split the getter out to it's own method - this makes it easier to override (see userbar code). 
```suggestion
  get targetContainer() {
      this.targetContainer = document.querySelector(
      this.targetValue,
    ) as HTMLElement;

    if (!this.targetContainer) {
      throw new Error('Target container not found');
    }
    }

  append() {
```